bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -356,13 +359,13 @@ protected void clearCandidates() {
         this.candidates = null;
     }
 
-    public ArrayList<E> getExpressionParts() {
+    public List<E> getExpressionParts() {
         return expressionParts;
```

Comment:
ack! see the setter -> I realize this was surprising, so I will change it to make it unmodifiable here instead of keeping an unmodifiable collection all the time...